src/commands/api/update.js

```diff
@@ -26,36 +40,39 @@ class UpdateAPICommand extends BaseCommand {
 
   async run() {
     const { args, flags } = this.parse(UpdateAPICommand)
-    const definition = parseDefinition(flags.file)
-    const apiVersion = getVersion(definition)
+    const definition = flags.file ? parseDefinition(flags.file) : null
     const requestedApiPath = getApiIdentifierArg(args)
-    const [owner, name, version = apiVersion] = splitPathParams(requestedApiPath)
+    const [owner, name, version] = splitPathParams(requestedApiPath)
+    const defaultVersion = definition ? getVersion(definition) : await this.getDefaultApiVersion([owner, name])
+    const apiVersion = version ? version : defaultVersion
 
     await this.executeHttp({
-      execute: () => getApi([owner, name, version]), 
-      onResolve: () => this.updateApi(owner, name, version, flags),
+      execute: () => getApi([owner, name, apiVersion]), 
+      onResolve: () => this.updateApi(owner, name, apiVersion, flags),
       options: { resolveStatus: [403] }
     })
     const apiPathWithVersion = requestedApiPath.split('/').length === 3 ?
     requestedApiPath :
-    `${requestedApiPath}/${apiVersion}`
+    `${requestedApiPath}/${defaultVersion}`
```

Comment:
Should this stay as `apiVersion`

Reply:
If requestedApiPath is doesn't have 3 variables it means the version is missing. So, `defaultVersion` and `apiVersion` are equal here but `defaultVersion` has a more relevant meaning.

Reply:
But if the requestedApiPath does have 3 variables we should take the version from there. This way we're only getting the default version or version from file.
I'm not sure why we're getting defaultVersion, is it in the ticket?